packages/cli/src/commands/org/config/monitor.ts

```diff
@@ -63,14 +66,38 @@ export default class Monitor extends Command {
   }
 
   private async monitorSnapshot(snapshot: Snapshot) {
-    cli.action.start(
-      `Operation: ${this.getReportType(snapshot.latestReport)}`,
-      this.getReportStatus(snapshot.latestReport)
-    );
-
+    const reporter = new SnapshotReporter(snapshot.latestReport);
+    cli.action.start(`Operation ${reporter.type}`, reporter.status);
     await snapshot.waitUntilDone(this.waitOption);
+    await this.displayMonitorResult(snapshot, reporter);
+  }
 
-    cli.action.stop(this.getReportStatus(snapshot.latestReport));
+  private async displayMonitorResult(
+    snapshot: Snapshot,
+    reporter: SnapshotReporter
+  ) {
+    if (!reporter.isSuccessReport()) {
+      await this.displaySnapshotError(snapshot, reporter);
+    }
+  }
+
+  // TODO: CDX-533: use Custom error instead
+  private async displaySnapshotError(
+    snapshot: Snapshot,
+    reporter: SnapshotReporter
+  ) {
+    cli.log(ReportViewerStyles.error(reporter.resultCode));
+    cli.log();
```

Comment:
🤨 why?

Reply:
Just to add some space and prevent stuff from being crowded.
I am addressing that in CDX-533